src/cli_parser.cpp

```diff
@@ -197,16 +194,7 @@ void CLIParser::parse(int argc, char** argv) {
         result = std::make_unique<cxxopts::ParseResult>(options->parse(argc, argv));
 
         if (result->unmatched().size()) {
-            std::cerr << "error parsing options - unmatched arguments: ";
-            for (auto& argument : result->unmatched()) {
-                std::cerr << argument << ", ";
-            }
-            std::cerr << std::endl;
-#ifdef __linux__
-        exit(EX_USAGE);
-#elif _WIN32
-        exit(3);
-#endif
+            this->graphOptionsParser.parse(result->unmatched());
```

Comment:
Wouldn't it be clearer to keep handling unmatched here & return the result from graphOptionsParser? You don't need tpo try catch in graphOptions parser then as well I think

Reply:
I think we can stick to the same flow in the parser classes.
Another thing is that once more options are implemented we can think about refactoring this code.